### DISCUSSION TITLE
Where to find official com.google.android.GoogleCamera without google ?

### DISCUSSION THREAD

**Discussion by tifriis208** [lukaspieper/Gcam-Services-Provider#17] [Q&A]
Hello, Thanks for you work with Gcam-Services-Provider.
I just bought a pixel 3 and installed grapheneOS on it. I looked for "Gcam" or "GoogleCamera" on Aurora but I cant' find it. 
Do you know where I can find this app on a safe way ?

Thanks :)

**Comment by innir** [comment]
You can actually pull it from the factory images google provides at https://developers.google.com/android/images

With the Pixel 4a December image as an example, that's what you need to do

```
$ unzip sunfish-rq1a.201205.008-factory-46217d49.zip
$ unzip sunfish-rq1a.201205.008/image-sunfish-rq1a.201205.008.zip product.img
$ simg2img product.img product.raw
$ sudo mount -o ro product.raw /mnt/
$ cp /mnt/app/GoogleCamera/GoogleCamera.apk .
$ sudo umount /mnt
```

**Comment by tifriis208** [comment]
Here the best way (and easiest) I found for my GrapheneOS: I got the apk (v7.4.201.322479879) from the [calyx-fdroid-repo](https://gitlab.com/CalyxOS/calyx-fdroid-repo). 

I got this issue (#6) with the apk from the factory images from google (11.0.0 (RQ1A.201205.003.A1, Dec 2020, Verizon)).

For information, Google Camera need Google Photo to display token pictures (#11)

**Reply by lukaspieper** [reply]
Thanks for sharing.

You didn't link the issues unfortunately, only the issue list. You can easily create a link to them using the id. (#6, #11).

**Reply by tifriis208** [reply]
Thanks, I edited my message 👍

**Reply by tifriis208** [reply]
Quite strange, I have the impression that some messages have disappeared.

Someone proposed to download the apk from apkmirror

**Reply by lukaspieper** [reply]
I've cleaned up the discussion (and title) a little to highlight the answers that I think are good/recommended.

**Reply by lukaspieper** [reply]
Looks like Calyx removed all proprietary apps from their repo ([source](https://gitlab.com/CalyxOS/calyx-fdroid-repo/-/commit/1d539c2ace6351962eb23c14588526f225753522)). So this is no longer a solution.

**Comment by lukaspieper** [comment]
Can anyone confirm that you can now install/update Google Camera via Aurora Store? It works for me with latest GrapheneOS. Possibly something to do with the recent changes to GrapheneOS regarding Play Services?

**Reply by innir** [reply]
Installing works for me too - but it's not stable and crashes often during app launch...

**Reply by innir** [reply]
For 8.2.400.* I get this error message:

    [...]
    java.lang.RuntimeException: android.view.InflateException: Binary XML file line #45 in com.google.android.GoogleCamera:layout/edu_bottomsheet: Binary XML file line #45 in com.google.android.GoogleCamera:layout/edu_bottomsheet: Error inflating class com.google.android.material.button.MaterialButton
    [...]
    Caused by: java.lang.RuntimeException: Can't create handler inside thread Thread[00UiWorker,5,main] that has not called Looper.prepare()
    [...]

No idea what causes this ...

**Reply by innir** [reply]
It works with Sandboxed Play services installed on GrapheneOS but not with Gcam-Services-Provider - the error doesn't look related though :-/

**Reply by gryban1** [reply]
The following behaviour is a stable workaround for the gcam 8.2.400:

Install googles app "photos"
Launch "photos"
now launch gcam
-> no crash and works

as long as you have photos launched in the background gcam works smoothly.
when you close photos in most cases you have to launch photos again before using gcam.

**Reply by lukaspieper** [reply]
There is now an [issue](https://github.com/lukaspieper/Gcam-Services-Provider/issues/54) for this crash.
